contracts/lib/CircularBufferUintLib.sol

```diff
@@ -0,0 +1,150 @@
+pragma solidity ^0.5.0;
+
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+/**
+ * @title Circular buffer for `uint`s.
+ *
+ * @notice This library represents a circular buffer that stores `uint`s. When
+ *         a set number of `uint`s have been stored, the storage starts
+ *         overwriting older entries. It overwrites always the oldest entry in
+ *         the buffer.
+ */
+library CircularBufferUintLib {
+
+    struct BufferUint {
+        /**
+         * The circular buffer that stores the latest `items.length` items. Once
+         * `items.length` items were stored, items will be overwritten starting at
+         * zero.
+         */
+        uint256[] items;
+
+        /**
+         * The current index in the items array. The index increases up to
+         * `items.length - 1` and then resets to zero in an endless loop. This
+         * means that a new item will always overwrite the oldest item.
+         */
+        uint256 index;
```

Comment:
If I correctly understand the purpose of applying this pattern, I would have expected `index` to be typed to something smaller. More specifically, I expected it to be`uint8`, but maybe I'm reading too much into the `MAX_STORAGE_ROOTS` in `GateywayBase` 🤔 

Or we want to allow maximum flexibility?

Reply:
Agree, `uint256` very big for this.
Since it's constant I agree it can be `uint8` (or `uint16`).